src/sql/parts/objectExplorer/common/objectExplorerViewTreeShim.ts

```diff
@@ -0,0 +1,152 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import { ITreeItem } from 'sql/workbench/common/views';
+import { IObjectExplorerService } from 'sql/parts/objectExplorer/common/objectExplorerService';
+import { ConnectionProfile } from 'sql/platform/connection/common/connectionProfile';
+import { ICapabilitiesService } from 'sql/platform/capabilities/common/capabilitiesService';
+import { TreeNode } from 'sql/parts/objectExplorer/common/treeNode';
+import { IConnectionManagementService, IConnectionDialogService, ConnectionType } from 'sql/platform/connection/common/connectionManagement';
+import { Deferred } from 'sql/base/common/promise';
+
+import { IConnectionProfile } from 'sqlops';
+
+import { TreeItemCollapsibleState } from 'vs/workbench/common/views';
+import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
+import { TPromise } from 'vs/base/common/winjs.base';
+import { equalsIgnoreCase } from 'vs/base/common/strings';
+import { hash } from 'vs/base/common/hash';
+import { generateUuid } from 'vs/base/common/uuid';
+
+export const SERVICE_ID = 'oeShimService';
+export const IOEShimService = createDecorator<IOEShimService>(SERVICE_ID);
+
+export interface IOEShimService {
+	_serviceBrand: any;
+	getChildren(node: ITreeItem, identifier: any): TPromise<ITreeItem[]>;
+	providerExists(providerId: string): boolean;
+}
+
+export class OEShimService implements IOEShimService {
```

Comment:
I have a question regarding the naming convention. Do we follow a 'class name should match file name' kind of practice - it would be better for code redability (findability?).